Build fails with upcoming Boost 1.87.0 due to Boost Asio removals

# Issue

Build fails with upcoming Boost 1.87.0 (tested beta1, stable release probably in December) due to Boost Asio removal of some deprecated APIs https://github.com/boostorg/asio/commit/a69e0e199706034d3e0ea5652afaf7df5741775d.
```
/tmp/osrm-backend-20241119-79427-bkjo50/include/server/server.hpp:56:46: error: expected ';' after expression
   56 |         boost::asio::ip::tcp::resolver::query query(address, port_string);
      |                                              ^
      |                                              ;
/tmp/osrm-backend-20241119-79427-bkjo50/include/server/server.hpp:56:9: error: no member named 'query' in 'boost::asio::ip::basic_resolver<boost::asio::ip::tcp>'; did you mean 'boost_asio_query_fn::query'?
   56 |         boost::asio::ip::tcp::resolver::query query(address, port_string);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         boost_asio_query_fn::query
```

This has been deprecated since Boost 1.66.0 https://github.com/boostorg/asio/commit/b60e92b13ef68dfbb9af180d76eae41d22e19356. Looking at Boost commit, something like following may work:
```diff
@@ -53,8 +53,7 @@ class Server
         const auto port_string = std::to_string(port);

         boost::asio::ip::tcp::resolver resolver(io_context);
-        boost::asio::ip::tcp::resolver::query query(address, port_string);
-        boost::asio::ip::tcp::endpoint endpoint = *resolver.resolve(query);
+        boost::asio::ip::tcp::endpoint endpoint = *resolver.resolve(address, port_string).begin();

         acceptor.open(endpoint.protocol());
 #ifdef SO_REUSEPORT
```

# Steps to reproduce

Please provide the steps required to reproduce your problem.
- Set up Boost 1.87.0-beta1 (https://github.com/boostorg/boost/releases/tag/boost-1.87.0.beta1)
- Build `osrm-backend`, either `master` branch or 5.27.1.

# Specifications

Please provide details of your development environment.
- Library/dependency versions - Homebrew dependencies
- Operating system - macOS 15 (local), Ubuntu 22.04 (CI)
- Hardware - M1 MacBook Pro (local), GCP e2-standard-8 (CI)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@
       - ADDED: Add support for opposite approach request parameter. [#6842](https://github.com/Project-OSRM/osrm-backend/pull/6842)
       - ADDED: Add support for accessing edge flags in `process_segment` [#6658](https://github.com/Project-OSRM/osrm-backend/pull/6658)
     - Build:
+      - CHANGED: Fix compilation with upcoming Boost 1.87.0 [#7073](https://github.com/Project-OSRM/osrm-backend/pull/7073)
       - CHANGED: Upgrade clang-format to version 15. [#6919](https://github.com/Project-OSRM/osrm-backend/pull/6919)
       - CHANGED: Use Debian Bookworm as base Docker image [#6904](https://github.com/Project-OSRM/osrm-backend/pull/6904)
       - CHANGED: Upgrade CI actions to latest versions [#6893](https://github.com/Project-OSRM/osrm-backend/pull/6893)

--- a/include/server/server.hpp
+++ b/include/server/server.hpp
@@ -53,8 +53,7 @@ class Server
         const auto port_string = std::to_string(port);
 
         boost::asio::ip::tcp::resolver resolver(io_context);
-        boost::asio::ip::tcp::resolver::query query(address, port_string);
-        boost::asio::ip::tcp::endpoint endpoint = *resolver.resolve(query);
+        boost::asio::ip::tcp::endpoint endpoint = *resolver.resolve(address, port_string).begin();
 
         acceptor.open(endpoint.protocol());
 #ifdef SO_REUSEPORT

